dependency-graph/service/graph/graph.go

```diff
@@ -0,0 +1,231 @@
+package Graph
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/mradulrathore/dependency-graph/service/node"
+)
+
+const (
+	NodeNotExistErr = "node doesn't exist, id:%d"
+	NodeExistMsg    = "node exists (id: %d)"
+)
+
+type Graph struct {
+	Nodes map[int]*node.Node
+}
+
+type GraphI interface {
+	GetParent(int) (map[int]*node.Node, error)
+	GetChild(int) (map[int]*node.Node, error)
+	GetAncestors(int) ([]int, error)
+	GetDescendants(int) ([]int, error)
+	DeleteEdge(int, int) error
+	DeleteNode(int) error
+	AddEdge(int, int) error
+	AddNode(int, string, map[int]string)
+	CheckIdExist(int) (*node.Node, bool)
+}
+
+func New() *Graph {
+	return &Graph{}
+}
+
+func (g *Graph) GetParent(id int) (map[int]*node.Node, error) {
+	n, exist := g.Nodes[id]
+	if !exist {
+		err := fmt.Errorf(NodeNotExistErr, id)
+		return nil, err
+	}
+
+	return n.Parent, nil
+}
+
+func (g *Graph) GetChild(id int) (map[int]*node.Node, error) {
+	n, exist := g.Nodes[id]
+	if !exist {
+		err := fmt.Errorf(NodeNotExistErr, id)
+		return nil, err
+	}
+
+	return n.Child, nil
+}
+
+func (g *Graph) GetAncestors(id int) ([]int, error) {
+	n, exist := g.Nodes[id]
+	if !exist {
+		err := fmt.Errorf(NodeNotExistErr, id)
+		return nil, err
+	}
+
```

Comment:
add unit test case for each method.

stick to TDD

Reply:
done!

Reply:
done!